test/verify/run_verify.cpp

```diff
@@ -88,10 +88,31 @@ inline void compile_check(migraphx::program& p,
     auto num = shapes.size();
     for(std::size_t i = 0; i < num; ++i)
     {
-        if(p.get_output_shapes()[i].lens() != shapes[i].lens())
+        auto output_shape = p.get_output_shapes()[i];
+        if(output_shape.dynamic() and shapes[i].dynamic())
```

Comment:
It seems it would be simpler to do `if(p.get_output_shapes()[i].to_dynamic().dyn_dims() != shapes[i].to_dynamic().dyn_dims())`.

Reply:
It would save a little code, but I like the error messages that distinguish the cases.